cmd/ctr/commands/run/run_unix.go

```diff
@@ -115,12 +124,42 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 			opts = append(opts, oci.WithImageConfig(image))
 			cOpts = append(cOpts,
 				containerd.WithImage(image),
-				containerd.WithSnapshotter(snapshotter),
+				containerd.WithSnapshotter(snapshotter))
+			if uidmap, gidmap := context.String("uidmap"), context.String("gidmap"); uidmap != "" && gidmap != "" {
+				cUID, hUID, uSize, err := parseRemapping(uidmap, "uidmap")
```

Comment:
since the only user of your function just needs a `spec.LinuxIDMapping`, I would suggest simplifying the `parseRemapping` function to just return that, and not have all the return types and then immediately create a struct. It also means you can probably get rid of `parseRemappingPart` and just call `parseRemapping` to grab the uid/gid for `TaskOpts` rather than having an extra function. The cost of parsing the whole map is negligible in this situation and simplifies the code to not have two slightly different parsing functions.

Reply:
Consolidated `parseRemappingPart()` into `parseIDMapping()` and only returned the `specs.LinuxIDMapping` struct.